language/move-prover/bytecode/src/read_write_set_analysis.rs

```diff
@@ -879,3 +842,73 @@ impl Default for ReadWriteSetState {
         }
     }
 }
+
+// =================================================================================================
+// Converting Infer Result into standalone types
+
+impl Offset {
+    pub fn normalize(&self, env: &GlobalEnv) -> Option<RWOffset> {
+        Some(match self {
+            Offset::Field(idx) => RWOffset::Field(*idx),
+            Offset::VectorIndex => RWOffset::VectorIndex,
+            Offset::Global(s) => RWOffset::Global(
+                s.get_type()
+                    .into_struct_type(env)
+                    .expect("Failed to normalize type"),
+            ),
+        })
+    }
+}
+impl AccessPath {
+    // `normalize` will return None when there's a malformed access path.
```

Comment:
Should this `expect` rather than returning an `Option`? I don't think malformed access paths should be possible unless their are bugs in the analysis.

Reply:
Ahhh fixed it in the other PR already!